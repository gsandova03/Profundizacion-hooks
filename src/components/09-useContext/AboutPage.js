import React from 'react'

export const AboutPage = () => {
  return (
    <div>
      <h4>About Page</h4>
      <hr/>
    </div>
  )
}
