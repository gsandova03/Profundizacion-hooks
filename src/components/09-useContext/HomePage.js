import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomePage = () => {

  const { user } = useContext( UserContext )
  console.log( user )
  return (
    <div>
      <h4>Home Page</h4>
      <hr/>
      <pre>
        { JSON.stringify( user, null, 3 ) }
      </pre>

    </div>
  )
}
