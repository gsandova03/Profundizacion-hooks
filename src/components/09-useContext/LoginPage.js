import React, { useContext } from 'react'
import { useForm } from '../../hooks/useForm'
import { UserContext } from './UserContext'

export const LoginPage = () => {

  const { setUser } = useContext( UserContext )
  return (
    <div>
      <h4>Login Page</h4>
      <hr/>
      <button 
        type="submit" 
        className="btn btn-primary"
        onClick={ ()=> setUser({
          id: 1242,
          name: "gabriel"
        })}
      > 
        Login 
      </button>
    </div>
  )
}
