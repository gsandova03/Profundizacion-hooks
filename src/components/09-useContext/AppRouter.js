import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { Navbar } from './components/Navbar';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/about" component={ AboutPage }/>
          <Route exact path="/login" component={ LoginPage }/>
          <Route exact path="/" component={ HomePage }/>
          <Redirect to="/"/>
        </Switch>
      </div>
    </Router>
  )
}
