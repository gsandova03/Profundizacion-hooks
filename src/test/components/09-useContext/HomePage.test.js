import { mount, shallow } from 'enzyme'
import React from 'react'
import { HomePage } from '../../../components/09-useContext/HomePage'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente <HomePage/>', () => {

  const user = {
    name:"gabriel",
    email:"gabriel@email.com"
  }
  //asignar el contex a nuestro componente

  //el mount es muy util cuando se quiere renderizar el componente completo y no solo el high order component
  const wrapper = mount(
    <UserContext.Provider value={{
      user
    }}>
      <HomePage/>
    </UserContext.Provider>
  )

  test('debe de mostrarse correctamente', () => {
    
    expect( wrapper ).toMatchSnapshot();

  })
  
})