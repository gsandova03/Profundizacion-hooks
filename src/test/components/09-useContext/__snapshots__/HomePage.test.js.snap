// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HomePage/> debe de mostrarse correctamente 1`] = `
<div>
  <h4>
    Home Page
  </h4>
  <hr />
  <pre>
    {
   "name": "gabriel",
   "email": "gabriel@email.com"
}
  </pre>
</div>
`;
